Criterion/Main/Options.hs

```diff
@@ -49,6 +49,11 @@ data MatchType = Prefix
                  -- @\"foo\"@ will match @\"foobar\"@.
                | Glob
                  -- ^ Match by Unix-style glob pattern.
+               | Pattern
+                 -- ^ Match by searching given substring in benchmark
+                 -- paths.
+               | IPattern
```

Comment:
The command-line help (see `parseWith`) should be updated to mention `pattern` and `ipattern`.

Moreover, it would also be a good idea to mention this in the tutorial. Currently, it just mentions `--match`, without any indication of what options you can actually _pass_ to `--match`. We should give brief descriptions of `prefix`, `glob`, `pattern`, and `ipattern` in that part of the tutorial as well.